src/classes/COFF_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,202 @@
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORSå
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Course Offering
+* @group-content ../../ApexDocContent/CourseOfferings.htm
+* @description Tests for COFF_CannotDelete_TDTM.
+*/
+@isTest
+private with sharing class COFF_CannotDelete_TEST {
+    /*********************************************************************************************************
+    * @description Test method to test the following:
+    * (1) If Prevent_Course_Offering_Deletion__c is enabled in Hierarchy Settings, and a Course Offering
+    * has child records, it cannot be deleted.
+    * (2) If Prevent_Course_Offering_Deletion__c is enabled in Hierarchy Settings, and a Course Offering
+    * does not have any child records, it can be deleted.
+    */
+    @isTest
+    private static void testPreventCourseOfferingDeletion() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                        (Account_Processor__c = UTIL_Describe_API.getHhAccRecTypeID(),
+                                                        Prevent_Course_Offering_Deletion__c = true));
+
+        Contact[] contacts = UTIL_UnitTestData_TEST.getMultipleTestContacts(6);
+        insert contacts;
+
+        List<Course_Offering__c> courseOfferings = new List<Course_Offering__c>();
+        Course_Offering__c courseOffering1 = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering2 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering3 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering4 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering5 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering6 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering7 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        courseOfferings.add(courseOffering1);
```

Comment:
Sorry noticed this just now, looks like we are inserting Course Offerings in UTIL_UnitTestdata_TEST, so I think we can remove adding these to the list courseOfferings.

Reply:
@skosgi1 , the reason why I'm adding them to the list "courseOfferings" is not for the insertion DML but because later on on line 121, I'm deleting all the Course Offerings. I didn't want to have to do multiple delete dmls for all 7 course offering records.

Reply:
Oops my bad!